common/src/main/java/org/apache/drill/common/util/DrillVersionInfo.java

```diff
@@ -49,10 +53,52 @@ public static String getVersion() {
         }
       }
     } catch (IOException except) {
-      appVersion = "Unknown";
+      appVersion = UNKNOWN_VERSION;
     }
     return appVersion;
   }
 
+  /**
+   * Compare two Drill versions disregarding build number and comparing only major and minor versions.
+   * Versions are considered to be compatible:
+   * 1. if current version is the same as version to compare.
+   * 2. if current version minor version + 1 is the same as version to compare.
+   */
+  public static boolean isVersionsCompatible(String currentVersion, String versionToCompare) {
+    if (currentVersion != null && currentVersion.equals(versionToCompare)) {
+      return true;
+    }
+
+    BigDecimal currentVersionDecimal = getVersionAsDecimal(currentVersion);
+    BigDecimal versionToCompareDecimal = getVersionAsDecimal(versionToCompare);
+
+    if (currentVersionDecimal != null && versionToCompareDecimal != null) {
```

Comment:
I guess, yes. Ideally all drillbits in cluster will be with the same version, only during rolling upgrades this situation may occur.